### DISCUSSION TITLE
HEI from ASA24 for multiple recalls

### DISCUSSION THREAD

**Discussion by csh38** [jamesjiadazhan/dietaryindex#196] [Q&A] [answered]
Hello all,

I ran dietaryrecall using the ASA24 totals file (26 participants, 3 recalls per participant) and it returned one HEI score per participant. How is this calculated? and is there a way to return HEI scores for each individual recall (so 3 per participant)?

Thanks!

**Comment by jamesjiadazhan** [comment, marked as answer]
Hi,

Thanks for using dietaryindex!

I just updated the package for you. As of version 1.0.3, the package can now generate flexible outputs for ASA24 for HEI2015 and HEI2020, depending on if RECALL_SUMMARIZE is TRUE or FALSE. If RECALL_SUMMARIZE is TRUE, all dietary recall results will be summarized via average. If not, all individual recall results will be shown. 

Check the manual page for more details:
https://github.com/jamesjiadazhan/dietaryindex/blob/main/Manual.md

If you are using the Dietaryindex package in your research, please be sure to cite our original work. By doing so, you not only add credibility to your findings but also recognize and appreciate our intellectual efforts and contributions. The appropriate citation is as follows:

- Jiada James Zhan, Rebecca A Hodge, Anne Dunlop, et al. Dietaryindex: A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies. bioRxiv. Published online August 07, 2023:2023.08.07.548466. doi:10.1101/2023.08.07.548466

Also, if possible, please give **dietaryindex** a STAR on GitHub (right top corner). It helps to spread out our work. 

Thanks! Let me know if you have other questions.

**Reply by csh38** [reply]
Hi!

Thank you for your help! 

I uninstalled the package and then reinstalled it in hopes of getting the updates version, but it still says I have version 1.0.2 and that it can't find the HEI2020_ASA24 function. Any tips for fixing this?

Error in HEI2020_ASA24(DATA_PATH, RECALL_SUMMARIZE = TRUE) : 
  could not find function "HEI2020_ASA24"

**Reply by csh38** [reply]
I resolved this issue!!!
Thanks again!

**Comment by jamesjiadazhan** [comment]
Sorry, I missed your first question. In the past, the ASA24 function summarizes all dietary recalls by doing average and there is no way to make it flexible. Now, it is flexible that you can ask it to summarize or not.
